stdlib/public/core/Codable.swift

```diff
@@ -81,8 +81,11 @@ public protocol CodingKey: Sendable,
 extension CodingKey {
   /// A textual representation of this key.
   public var description: String {
-    let intValue = self.intValue?.description ?? "nil"
-    return "\(type(of: self))(stringValue: \"\(stringValue)\", intValue: \(intValue))"
+    if let intValue {
+      "[" + String(intValue) + "]"
```

Comment:
I think `"[\(intValue)]" is more idiomatic and produces fewer instructions too.

Reply:
I wrote it the way I did because moving brackets in with quotes and interpolation is always kinda messy, but your argument makes sense. I'm guessing small optimizations are worth it in stdlib code!